### PR TITLE
fix: jbs-e2e: use up-to-date test image

### PR DESCRIPTION
**Pull request by psturc**
the reference to the root (test) image was outdated

### CODE REVIEW

**Comment by psturc** [comment]
/pj-rehearse pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-e2e

File: ci-operator/config/redhat-appstudio/jvm-build-service/redhat-appstudio-jvm-build-service-main.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 build_root:
   image_stream_tag:
-    name: redhat-appstudio-e2e-test-runner
+    name: redhat-appstudio
     namespace: ci
-    tag: latest
+    tag: e2e-test-runner
```
Comment (psturc) [inline ci-operator/config/redhat-appstudio/jvm-build-service/redhat-appstudio-jvm-build-service-main.yaml:5]: this was an old image used at the time when e2e-tests image was based on go v1.18 and it was not updated after we moved to go v1.19

thus we should be using the image that's being updated by [e2e-tests job](https://github.com/openshift/release/blob/ac05fd6d3bebebff8c8d3e6fefbb881651b0b94d/ci-operator/config/redhat-appstudio/e2e-tests/redhat-appstudio-e2e-tests-main.yaml#L22-L25)

**Comment by psturc** [comment]
@stuartwdouglas for testing purposes I used [my branch](https://github.com/redhat-appstudio/jvm-build-service/pull/718) that is based on [your PR](https://github.com/redhat-appstudio/jvm-build-service/pull/708) and the CI has passed

@gabemontero @stuartwdouglas feel free to add /lgtm if you're happy with the change, thanks

**Comment by stuartwdouglas** [comment]
I don't have permissions unfortunately

**Comment by stuartwdouglas** [comment]
@psturc those tests should pass now if you re-run them

**Comment by psturc** [comment]
ok thanks I'll try to rerun them. they should not be required to pass in this PR though - it's stuck in "merge pool" for some reason
Screenshot 2023-07-18 at 10 41 41

/pj-rehearse pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-jbs-e2e pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-quickstarts-e2e

**Comment by psturc** [comment]
/pj-rehearse pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-jbs-e2e

**Pull request merged**